### PR TITLE
Create Blog “explore-the-hpe-greenlake-for-compute-ops-management-rest-api-using-python-and-powershell”

### PR DESCRIPTION
**Pull request by pattdepanick**
Automatically generated by Netlify CMS

### CODE REVIEW

**Review by choukden — CHANGES_REQUESTED** [review]
Merci Fredo

Just a few minor comments.

Well done!!!

Denis

File: content/blog/explore-the-hpe-greenlake-for-compute-ops-management-rest-api-using-python-and-powershell.md [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+author: Vincent Berger & Frederic Passeron
+authorimage: /img/vb-fp192.png
+disable: false
+tags:
+  - "hpe-greenlake "
+  - "hpe-greenlake-for-compute-ops-management "
+  - "developer "
+  - devops
+---
+HPE GreenLake for Compute Ops Management automates and transforms complex and time-consuming compute management operations into a simplified experience across edge-to-cloud. Whether you are an IT OPS or a DEV OPS engineer, you know that automation is the key to success. And today’s automation relies heavily on APIs and how one can interact easily with them. So, let us show you how to leverage the API provided so you, too, can take advantage of what HPE GreenLake for Compute Ops Management can provide.
```
Comment (drensing) [inline content/blog/explore-the-hpe-greenlake-for-compute-ops-management-rest-api-using-python-and-powershell.md:14]: Seeing as how this introduction is exactly the same as the one in your other blog post, perhaps you should add a line at the end of each of these that shows the difference between the two blog posts. For instance, for this one, consider adding in "In this blog post, we will cover how to do this using Python and PowerShell." and for the other, add in a line at the end of the first paragraph that says "In this blog post, we will cover how to do this using cURL and Postman."

File: content/blog/explore-the-hpe-greenlake-for-compute-ops-management-rest-api-using-python-and-powershell.md [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+* Obtain the list of servers in your account 
+* Obtain the list of firmware bundles
+* Obtain the list of job templates
+* Perform a firmware update
+* Run a carbon emissions report
+* Power on/off/restart a server
+
+![figure2](/img/greenlake-com-ops-api-pyt-pow9.png)
+
+I﻿n a previous [blog](https://developer.hpe.com/blog/explore-the-hpe-greenlake-for-compute-ops-management-rest-api-using-curl-and-postman), we covered some of basics of this API using cURL or Postman. We would like to share with you today a few examples of REST API calls that can be made through simple Python and PowerShell codes examples
```
Comment (drensing) [inline content/blog/explore-the-hpe-greenlake-for-compute-ops-management-rest-api-using-python-and-powershell.md:35]: Recommend you rephrase the second sentence to:
In this post, we would like to share with you a few examples of REST API calls that can be made through simple Python and PowerShell code examples.
(note "code" versus "codes", the addition of the period at the end of the sentence, and the change at the beginning - since your today could easily be their tomorrow :-) )

File: content/blog/explore-the-hpe-greenlake-for-compute-ops-management-rest-api-using-python-and-powershell.md [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+There are several ways to invoke the API:
+
+* Using REST directly or from JavaScript 
+* Using cURL
+* Using Python
+* Using Go
+* Using PowerShell
+* Using Ansible, Terraform, etc.
+
+The [HPE GreenLake for Compute Ops Management API Reference site](https://developer.greenlake.hpe.com/docs/greenlake/services/compute-ops/public/openapi/compute-ops-latest/overview/) leverages an OpenAPI conformant documentation that provides a complete explanation of the operations supported by the Unique Resource Identifiers [(URIs)](<>)as well as sample requests and responses.
```
Comment (drensing) [inline content/blog/explore-the-hpe-greenlake-for-compute-ops-management-rest-api-using-python-and-powershell.md:55]: Reading this in the preview, I think I better understand a little something here. Please take out "an" prior to "OpenAI conformant" and place a hyphen between "OpenAPI" and "conformant" so it reads "...leverages OpenAPI-conformant documentation that..."

Also note, there's a space required after (URIs). See snip:

image

File: content/blog/explore-the-hpe-greenlake-for-compute-ops-management-rest-api-using-python-and-powershell.md [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+
+# Using PowerShell
+
+From a Terminal application, I start a PowerShell environment and set some variables that will be used later.
```
Comment (drensing) [inline content/blog/explore-the-hpe-greenlake-for-compute-ops-management-rest-api-using-python-and-powershell.md:59]: Recommend you reword line 59 to "From a terminal application, start a PowerShell environment and set the variables that will be used later."

File: content/blog/explore-the-hpe-greenlake-for-compute-ops-management-rest-api-using-python-and-powershell.md [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+$endpoint = "https://us-west2-api.compute.cloud.hpe.com"
+# Initialize an empty headers array
+$headers = @{} 	
+# Add a header for authorization, using the Access Token generated in task 2	
+$headers["Authorization"] = "Bearer_access_token_here"
+```
+
+![figure3](/img/greenlake-com-ops-api-pyt-pow1.png)
+
+Our first request:
```
Comment (drensing) [inline content/blog/explore-the-hpe-greenlake-for-compute-ops-management-rest-api-using-python-and-powershell.md:72]: Recommend you reword this to "For the first request, try:"

File: content/blog/explore-the-hpe-greenlake-for-compute-ops-management-rest-api-using-python-and-powershell.md [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+
+When the request is successful, the response should display 200:
+
+![figure4](/img/greenlake-com-ops-api-pyt-pow2.png)
+
+```powershell
+$response.StatusCode
+```
+
+The JSON data returned by the request is in $response.Content but it is rather hard to read as is, I can prettify it a bit with:
```
Comment (drensing) [inline content/blog/explore-the-hpe-greenlake-for-compute-ops-management-rest-api-using-python-and-powershell.md:86]: There's a space missing between the period and the next sentence. And, the next sentence could read better:
"The JSON data returned by the request is in $response. But the content is rather hard to read as is, so you might want to prettify it a bit with:"
image

File: content/blog/explore-the-hpe-greenlake-for-compute-ops-management-rest-api-using-python-and-powershell.md [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+
+Now, the request returned 2 objects (count : 2) and the actual server details are in the items array property.
+
+![figure5](/img/greenlake-com-ops-api-pyt-pow3.png)
+
+The API supports several query parameters depending on the resource type, such as filter, limit (maximum number of records to return), offset (resource offset to start the response from) and sort (order in which to return the resources in the collection).
+
+Filters provide the ability to limit the resources that take part in the action of a REST call. When a REST call includes a filter, the GET or DELETE action is restricted to a response that meets the filter requirements. Filters are specified by using the query parameter 'filter'.
+
+A simple example of filtering follows:
```
Comment (drensing) [inline content/blog/explore-the-hpe-greenlake-for-compute-ops-management-rest-api-using-python-and-powershell.md:100]: I prefer how you worded it in the other blog post: "Here is a simple example of filtering:"

File: content/blog/explore-the-hpe-greenlake-for-compute-ops-management-rest-api-using-python-and-powershell.md [hunk 8]
```diff
@@ -0,0 +1,242 @@
...
+
+**GET <URI>?filter=powerState eq 'Off'**
+
+This example shows a simple filter. The resources returned by the query are limited to results with the attribute powerState and the value Off.
```
Comment (drensing) [inline content/blog/explore-the-hpe-greenlake-for-compute-ops-management-rest-api-using-python-and-powershell.md:104]: If you reword it, as I suggest above, then you don't need this first sentence in 104 and you can start line 104 with "The resources returned by the query are..."

The reason I prefer it this way is because it's always better to tell the reader up front what you're going to do as opposed to doing something and then explaining what it was. The human brain reads and understands (in general... but not in all languages) top to bottom.

File: content/blog/explore-the-hpe-greenlake-for-compute-ops-management-rest-api-using-python-and-powershell.md [hunk 9]
```diff
@@ -0,0 +1,242 @@
...
+$headers
+```
+
+I start by checking the request was successful as before:
+
+```powershell
+$response.StatusCode
+```
+
+it should display 200
```
Comment (drensing) [inline content/blog/explore-the-hpe-greenlake-for-compute-ops-management-rest-api-using-python-and-powershell.md:123]: This line should read "It should display 200." (period at the end and initial capital letter at the beginning of the sentence)

File: content/blog/explore-the-hpe-greenlake-for-compute-ops-management-rest-api-using-python-and-powershell.md [hunk 10]
```diff
@@ -0,0 +1,242 @@
...
+print(json.dumps(response.json(), indent=2))
+```
+
+![figure 11](/img/greenlake-com-ops-api-pyt-pow8.png)
+
+Refer to previous sections for more information on filter query syntax and API references.
+
+## What’s next?
+
+In this blog post, we covered how to get started with HPE GREENLAKE FOR COMPUTE OPS MANAGEMENT REST API explained how to post simple API calls through Python and PowerShell. Some future Workshops-on-Demand will allow you to run experiment these examples live, stay tuned.
```
Comment (drensing) [inline content/blog/explore-the-hpe-greenlake-for-compute-ops-management-rest-api-using-python-and-powershell.md:234]: "HPE GreenLake for Compute Ops Management REST API" should not be in ALL CAPS. It also needs the word "and" between it an "explained how to post simple API calls..."

The very last sentence needs a rewrite. Perhaps this would do?
"In some of our future Workshops-on-Demand, you will have the opportunity to run live experiments on these examples. Stay tuned for more details!"

I also recommend you then hotlink to the "Workshops-on-Demand" catalog.

**Review by drensing — COMMENTED** [review]
Here you go, Frederic... my comments on the second blog post.
